src/components/SignUp/CareerInfo.js

```diff
@@ -74,6 +75,20 @@ class CareerInfo extends Component {
       Actions.refresh({ rightTitle: 'SAVE', onRight: this.onNextBtnPressed.bind(this) });
   }
 
+  componentWillReceiveProps(props) {
+    if(props.fromEdit && this.state.needRefresh) {
```

Comment:
If it is true, It means this component has been called from editProfile component
(Otherwise, It must be called from sign up flow)